Edits

timohanke opened the pull request:
Various code improvements

timohanke committed:
Remove unused type

timohanke committed:
Remove unnecesary line in with expression

timohanke committed:
Remove private keywords

timohanke committed:
Remove return keyword in query functions and one-line functions

timohanke committed:
Make do ? a one-liner conforming to other places in the code

timohanke committed:
Move base64 image code to separate file

timohanke committed:
Put all assertions into a module

* the namespace is convenient to read
* we can collectively comment on them by commenting on the module

timohanke committed:
Replace a switch statement by shorter do ? block

timohanke committed:
inline the negation of isBase64Char

Just for completion, in case someone wants to use this code in
another place where performance matters

timohanke committed:
Reduce nesting levels in validateImage

Move looking at data to outside of the for loop

timohanke committed:
Combine multiple ignore do ? blocks into one

timohanke committed:
Group some statements mainly in actor init sequence

timohanke committed:
Shorten type names and related variable names

reginleif888 commented on src/base64.mo:
```diff
@@ -0,0 +1,43 @@
+    (c >= 'A' and c <= 'Z') or (c >= 'a' and c <= 'z') or (c >= '0' and c <= '9') or c == '+' or c == '/' or c == '=';
+  };
+
+  func isNotBase64(c : Char) : Bool {
```
Does it make sense to duplicate the logic for a negative check if we can use `not isBase64(char)`?

timohanke replied:
For our application it is not needed. I only wanted to have the code for both conditions (positive and negative) written down and checked in so that others can copy-paste the code if they need it.

If someone has an application that is more sensitive to performance than ours, for example for very large files, then they will want to avoid to wrap the function call. Also the negative test has 7 comparisons vs 9 for the positive test. So they will prefer the direct negative test.

reginleif888 replied:
Got it, thanks

timohanke committed:
Add hand-crafted, documented .did file

timohanke committed:
Make symbol/name size limits of 8/64 inclusive, not exclusive

timohanke committed:
Restrict character sets for symbol and names

This is to avoid mistakes. The character sets can be relaxed later through an upgrade.

timohanke committed:
Remove character set restriction on name again

Because the name can be updated there is no risk of irreversible mistakes

timohanke committed:
Mention symbol character set in did file
